training/analysis_functions.py

```diff
@@ -0,0 +1,48 @@
+from typing import Union, List, TypeVar
+
+import pandas as pd
+import numpy as np
+import pandas.core.frame
+import pandas.core.series
+
+DataFrameType = TypeVar("pandas.core.frame.DataFrame")
```

Comment:
nie wiem, czy to jest potrzebne, ale jeśli już to chyba lepiej `DataFrameType = NewType("DataFrameType", pandas.core.frame.DataFrame)`